kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/service/KafkaConnectService.java

```diff
@@ -0,0 +1,227 @@
+import reactor.core.publisher.Mono;
+import reactor.retry.Retry;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class KafkaConnectService {
+    private static final int MAX_RETRIES = 5;
+
+    private final ClustersStorage clustersStorage;
+    private final ClusterMapper clusterMapper;
+    private final KafkaConnectMapper kafkaConnectMapper;
+
+    public Mono<Flux<Connect>> getConnects(String clusterName) {
+        return Mono.just(
+                Flux.fromIterable(clustersStorage.getClusterByName(clusterName)
+                        .map(KafkaCluster::getKafkaConnect).stream()
+                        .flatMap(Collection::stream)
+                        .map(clusterMapper::toKafkaConnect)
+                        .collect(Collectors.toList())
+                )
+        );
+    }
+
+
+    public Flux<String> getConnectors(String clusterName, String connectName) {
+        return getConnectAddress(clusterName, connectName)
+                .flatMapMany(connect -> withRetryOnConflict(
+                        KafkaConnectClients.withBaseUrl(connect).getConnectors()
+                        ).doOnError(log::error)
+                );
+    }
+
+    public Mono<Connector> createConnector(String clusterName, String connectName, Mono<NewConnector> connector) {
+        return getConnectAddress(clusterName, connectName)
+                .flatMap(connect ->
+                        connector.cache()
+                                .map(kafkaConnectMapper::toClient)
+                                .flatMap(c ->
+                                        withBadRequestErrorHandling(
+                                                withRetryOnConflict(
+                                                        KafkaConnectClients.withBaseUrl(connect).createConnector(c)
+                                                )
+                                        )
+                                )
+                                .flatMap(c -> getConnector(clusterName, connectName, c.getName()))
+                );
+    }
+
+    public Mono<Connector> getConnector(String clusterName, String connectName, String connectorName) {
+        return getConnectAddress(clusterName, connectName)
+                .flatMap(connect -> withRetryOnConflict(KafkaConnectClients.withBaseUrl(connect).getConnector(connectorName))
+                        .map(kafkaConnectMapper::fromClient)
+                        .flatMap(connector ->
+                                withRetryOnConflict(KafkaConnectClients.withBaseUrl(connect).getConnectorStatus(connector.getName()))
+                                        .map(connectorStatus -> {
+                                            var status = connectorStatus.getConnector();
+                                            return connector.status(kafkaConnectMapper.fromClient(status));
```

Comment:
Am I right that here you mutate the Connector object?

Reply:
Thanks, fixed